opt: Fix various name resolution bugs in optimizer

rytaft opened the pull request:
This commit fixes a couple of bugs in the optbuilder:

(1) The optbuilder was incorrectly allowing queries such
    as `SELECT * from a, a`. This query now throws the error:
    `cannot join columns from the same source name "a"`
    `(missing AS clause)`.

(2) The optbuilder was incorrectly causing an error for
    queries such as `SELECT t.kv.k FROM kv` when the current
    database was set to t. This is now fixed.

In order to test the fix to these bugs and add other test cases,
this commit also improves the optimizer test infrastructure to
qualify table names with the schema and catalog.

This commit also adds additional test cases identified by
@knz in a document he created describing tricky semantic
analysis functionality in CockroachDB.

Release note: None

rytaft committed:
opt: Fix various name resolution bugs in optimizer

This commit fixes a couple of bugs in the optbuilder:

(1) The optbuilder was incorrectly allowing queries such
    as `SELECT * from a, a`. This query now throws the error:
    `cannot join columns from the same source name "a"`
    `(missing AS clause)`.

(2) The optbuilder was incorrectly causing an error for
    queries such as `SELECT t.kv.k FROM kv` when the current
    database was set to t. This is now fixed.

In order to test the fix to these bugs and add other test cases,
this commit also improves the optimizer test infrastructure to
qualify table names with the schema and catalog.

This commit also adds additional test cases identified by
@knz in a document he created describing tricky semantic
analysis functionality in CockroachDB.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/24851)
<!-- Reviewable:end -->

knz commented:
Thanks for this!

I didn't review the correctness of the overall approach (but it does look fine at a high level) - letting others chime in.

The construction of the error messages however need to use the proper code path, suggestions below.

---

Reviewed 16 of 16 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

*[pkg/sql/opt/optbuilder/join.go, line 97 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFd7AL8UHTMWIIknaR:-LAFd7AL8UHTMWIIknaS:b-ji4o4n) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/join.go#L97)):*
> ```Go
> 		}
> 		if _, ok := leftTables[t.FQString()]; ok {
> 			panic(errorf("cannot join columns from the same source name %q (missing AS clause)", t.TableName))
> ```

`tree.ErrString(&t.TableName)`? Or probably `tree.ErrString(t)` (if `rightCol.table` is a pointer already. Otherwise, perhaps you want `t := &rightCol.table` above)

---

*[pkg/sql/opt/optbuilder/testdata/project, line 290 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFbovB9kmzczVqEbAz:-LAFbovB9kmzczVqEbB-:b68sb66) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/testdata/project#L290)):*
> ```
>  ├── columns: k:1(int!null)
>  ├── scan kv
>  │    └── columns: kv.k:1(int!null) kv.v:2(int)
> ```

If you now support a catalog prefix, the plan dumps should report that prefix as well. Here and in every other test.

---

*[pkg/sql/opt/testutils/alter_table.go, line 43 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFcmVT9exViwub4Uz2:-LAFcmVT9exViwub4Uz3:b-tg3o06) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/testutils/alter_table.go#L43)):*
> ```Go
> 	table, ok := tc.tables[tn.FQString()]
> 	if !ok {
> 		panic(fmt.Sprintf("cannot find table %s", tn.Table()))
> ```

To be correct with the current choice to only have the unqualified table name: %q with `tree.ErrString(&tn.TableName)`

But probably you really want to use %q with `tree.ErrString(tn)` (qualified in message)

---

*[pkg/sql/opt/testutils/test_catalog.go, line 86 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFdOvXD539hyzdbtFP:-LAFdOvXD539hyzdbtFQ:bjkruvj) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/testutils/test_catalog.go#L86)):*
> ```Go
> 		return table, nil
> 	}
> 	return nil, fmt.Errorf("table %q not found", name.String())
> ```

table %q ... with `tree.ErrString(name)`

---

*[pkg/sql/opt/testutils/test_catalog.go, line 94 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFcb0Z2pb7LtbXMrdI:-LAFcb0Z2pb7LtbXMrdJ:b-15cyed) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/testutils/test_catalog.go#L94)):*
> ```Go
> 	tab, err := tc.FindTable(context.TODO(), &tn)
> 	if err != nil {
> 		panic(fmt.Errorf("table %s is not in the test catalog", name))
> ```

%q with `tree.ErrString(*(*tree.Name)&name)`

---

*[pkg/sql/opt/testutils/test_catalog.go, line 162 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFcNp350DiziIfGB7F:-LAFcNp350DiziIfGB7G:b-ao8ms8) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/testutils/test_catalog.go#L162)):*
> ```Go
> 		}
> 	}
> 	panic(fmt.Sprintf("cannot find column %s in table %s", name, tt.Name.String()))
> ```

1) %q with tree.ErrString((*tree.Name)&name)

2) %q with `tt.Name.String()` -> `tree.ErrString(&tt.Name)`

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-:-LAFdVykD16hPcM0Qdya:b-igt1xm)*
<!-- Sent from Reviewable.io -->

andy-kimball commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions, some commit checks failed.

---

*[pkg/sql/opt/optbuilder/join.go, line 85 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAJ1R--8C6KEEvYecET:-LAJ1R--8C6KEEvYecEU:b-4g97hb) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/join.go#L85)):*
> ```Go
> // join.
> func (b *Builder) validateJoinTableNames(
> 	leftTables map[string]struct{}, rightScope *scope,
> ```

Wouldn't it be possible to pass a `[]scopeColumn` slice instead of the `leftTables` map? This would make the callers simpler (because they don't need to allocate and fill out a map), as well as avoiding object creation (i.e. better perf).

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851)*
<!-- Sent from Reviewable.io -->

knz commented:
Reviewed 6 of 6 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/opt/optbuilder/testdata/project, line 290 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFbovB9kmzczVqEbAz:-LAJwhdZDTKx2gWimgyZ:bl8pi3x) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/testdata/project#L290)):*
<details><summary><i>Previously, rytaft wrote…</i></summary><blockquote>

Unfortunately this would be a bit complicated. Once we build the memo, we no longer store any information about how the columns were qualified in the query. We instead access the column names from opt.Metadata, which stores one label for each column in the table. We could qualify the table names in these column labels, but that would match the qualification of the table name in the FROM clause, not the qualification of the columns themselves.  For example,

`SELECT k FROM t.kv` would output k as `t.public.kv.k`, whereas
`SELECT t.kv.k FROM kv` would output k as just `kv.k`

I assume this is not what you're suggesting... but in order to print the columns as they actually appear in the query we'd need to make significant changes to the way metadata is stored and accessed.

For example,

`SELECT k, t.kv.k FROM t.kv` would require that we store k twice in the metadata with two different column labels.  I don't think it's worth the overhead just to improve the testing output.

Please let me know if I'm misunderstanding something. Thanks!

</blockquote></details>

I think we do not need to keep the original level of qualification.
However, in order to disambiguate the textual representation of the plan, I think it would be useful (IMHO *necessary*) to fully qualify all the column references in the printout.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851)*
<!-- Sent from Reviewable.io -->

rytaft commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/sql/opt/optbuilder/testdata/project, line 290 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFbovB9kmzczVqEbAz:-LAJxG0U--kTiqAKTb9R:b1g2yte) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/testdata/project#L290)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I think we do not need to keep the original level of qualification.
However, in order to disambiguate the textual representation of the plan, I think it would be useful (IMHO *necessary*) to fully qualify all the column references in the printout.
</blockquote></details>

So you'd suggest printing `t.public.kv.k` regardless of the original qualification of kv? Or only when the table is qualified (like in the first example above)?

My main issue with printing it all the time is it would add a lot of clutter to the test files. If we can remove the qualification by removing it from the query I think it would be fine. I can make that change pretty quickly.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851)*
<!-- Sent from Reviewable.io -->

knz commented:
To check you have all the name resolution tests you need, you'd need to translate the unit tests in `sql/sem/tree/name_resolution_test.go` to logic tests, or outright port those unit tests to the `opt` package. A reasonable amount of de-duplication of the code would be desirable though, because we can't exclude we'll need to extend this in the future.

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/sql/opt/optbuilder/testdata/project, line 290 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFbovB9kmzczVqEbAz:-LAJzZDK1MZEEjy5Y5aw:bn03y32) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/testdata/project#L290)):*
<details><summary><i>Previously, rytaft wrote…</i></summary><blockquote>

So you'd suggest printing `t.public.kv.k` regardless of the original qualification of kv? Or only when the table is qualified (like in the first example above)?

My main issue with printing it all the time is it would add a lot of clutter to the test files. If we can remove the qualification by removing it from the query I think it would be fine. I can make that change pretty quickly.
</blockquote></details>

Yes that is my suggestion.
The problem that I want to solve is to disambiguate the printout when we join tables across separate databases, e.g. `select * from a.public.t, b.public.t`. I'd be sad to show the plan for that and see a join of `t` with itself.

Please open a followup issue. This should not block this PR.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-:-LAJzx_r2adnlCQ1hNGL:b-p5v0lj)*
<!-- Sent from Reviewable.io -->

andy-kimball commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/sql/opt/optbuilder/join.go, line 85 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAJ1R--8C6KEEvYecET:-LAK8C-a3aLHciok7pgt:b-276asg) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/join.go#L85)):*
<details><summary><i>Previously, rytaft wrote…</i></summary><blockquote>

Definitely possible, but the reason I did it this way was to only iterate through each set of columns once.  If we are joining two tables that each have 100+ columns, the nested loops algorithm could be expensive. And each time we join with a new table we would need to search all the existing columns again.

Another alternative is to store a map or slice of table names in the scope itself (instead of searching for table names inside the scopeColumns). I'm not sure if that would be better or worse... We can just add the table names directly from `buildScan`, but then we'd need to maintain the map/slice for each scope, even if it was never needed.

What do you think?
</blockquote></details>

I see. OK, this seems fine then. I didn't consider the many join/many cols case.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851)*
<!-- Sent from Reviewable.io -->

rytaft committed:
opt: print table qualification of column names in expression view output

If a table name is qualified with schema and/or catalog in a data-driven
test query, the columns from that table will also be qualified with
the schema and/or catalog in the expression view output.

This will ensure that for queries such as `SELECT * FROM db1.a, db2.a`,
it will be clear which columns are from db1.a and which columns are from
db2.a. Previously, it would look like all columns were from the same table
"a".

One can also choose to override this behavior using the flag
`fully-qualify-names`. If set, all column names in the output will be fully
qualified.

Release note: None

knz commented:
Reviewed 5 of 10 files at r3.
Review status: 17 of 22 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/sql/opt/optbuilder/testdata/project, line 290 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFbovB9kmzczVqEbAz:-LAKK43Y2C_zp9yEmEos:b5t28wn) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/testdata/project#L290)):*
<details><summary><i>Previously, rytaft wrote…</i></summary><blockquote>

I had already made most of the changes so I just added a new commit to this PR. Since the goal is just to disambiguate `a.public.t` from `b.public.t`, I think we only need to print the qualification for the columns if the tables themselves are qualified. Please take a look and let me know if you agree. (There is a relevant test case at the end of `optbuilder/testdata/inner-join`)
</blockquote></details>

well it's not exactly sufficient either. Consider this:

```
SELECT * FROM pg_type
```

vs. this:

```
CREATE TABLE pg_type(x INT);
SELECT * FROM pg_type
```

Both these queries would have the same printout, although they use completely different tables. That is still a problem.


Also I am not too fond of paying the price of allocating a special table label in the common case (i.e. for _every_ table reference in every query) although it is only used in rare cases (EXPLAIN).

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: 17 of 22 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/sql/opt/optbuilder/testdata/project, line 290 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFbovB9kmzczVqEbAz:-LAKLOMr4ljBePz3-urC:b1icycm) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/testdata/project#L290)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

well it's not exactly sufficient either. Consider this:

```
SELECT * FROM pg_type
```

vs. this:

```
CREATE TABLE pg_type(x INT);
SELECT * FROM pg_type
```

Both these queries would have the same printout, although they use completely different tables. That is still a problem.


Also I am not too fond of paying the price of allocating a special table label in the common case (i.e. for _every_ table reference in every query) although it is only used in rare cases (EXPLAIN).

</blockquote></details>

sorry in my latter example you also need to do `SET search_path=public, pg_catalog`.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851)*
<!-- Sent from Reviewable.io -->

rytaft commented:
Review status: 17 of 22 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[pkg/sql/opt/optbuilder/testdata/project, line 290 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFbovB9kmzczVqEbAz:-LAOAbHYEqWJ_8PMLNqQ:b38lvus) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/testdata/project#L290)):*
These tests are much simpler than the standard logic_test tests. Since we're not starting up a cluster, we don't have access to all the virtual schemas/tables (the first example you gave above would throw an error -- table pg_type not found). The way I wrote this PR, any unqualified table is automatically assumed to be part of catalog "t" and schema "public". So if I follow your suggestion and qualify every column name, the test output will be littered with "t.public...", making it harder to read. I do think we should test the optimizer with virtual schemas, but I don't think this is the right place. Most of the tests that rely on `test_catalog.go` and the other code updated by this PR are not focused on name resolution.

One option is to add a test directive to set a flag to fully qualify all column names. That way I can use the directive only for tests that are explicitly testing name resolution. Does that seem like a good solution?

> Also I am not too fond of paying the price of allocating a special table label in the common case (i.e. for every table reference in every query) although it is only used in rare cases (EXPLAIN).

I think @andy-kimball made the decision to include the column labels in metadata to allow printing and debugging (for tests like these). Andy, could you respond to Raphael's concern here?

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851)*
<!-- Sent from Reviewable.io -->

rytaft commented:
Review status: 12 of 22 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/opt/optbuilder/testdata/project, line 290 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFbovB9kmzczVqEbAz:-LAoFJ9fCixlma2pmtHk:biejosv) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/testdata/project#L290)):*
<details><summary><i>Previously, rytaft wrote…</i></summary><blockquote>

These tests are much simpler than the standard logic_test tests. Since we're not starting up a cluster, we don't have access to all the virtual schemas/tables (the first example you gave above would throw an error -- table pg_type not found). The way I wrote this PR, any unqualified table is automatically assumed to be part of catalog "t" and schema "public". So if I follow your suggestion and qualify every column name, the test output will be littered with "t.public...", making it harder to read. I do think we should test the optimizer with virtual schemas, but I don't think this is the right place. Most of the tests that rely on `test_catalog.go` and the other code updated by this PR are not focused on name resolution.

One option is to add a test directive to set a flag to fully qualify all column names. That way I can use the directive only for tests that are explicitly testing name resolution. Does that seem like a good solution?

> Also I am not too fond of paying the price of allocating a special table label in the common case (i.e. for every table reference in every query) although it is only used in rare cases (EXPLAIN).

I think @andy-kimball made the decision to include the column labels in metadata to allow printing and debugging (for tests like these). Andy, could you respond to Raphael's concern here?
</blockquote></details>

Just a gentle ping - 

@knz, does what I wrote above seem reasonable? 

@andy-kimball, do you agree with @knz that storing the table+column name for each column in the metadata is too much overhead? If so, we should address that in a separate PR.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851)*
<!-- Sent from Reviewable.io -->

knz commented:
Review status: 12 of 22 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/opt/optbuilder/testdata/project, line 290 at r1](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFbovB9kmzczVqEbAz:-LAoHJvO2dXS_bQfSbwg:b-2m3b52) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/testdata/project#L290)):*
> to add a test directive to set a flag to fully qualify all column names. That way I can use the directive only for tests that are explicitly testing name resolution. Does that seem like a good solution?

Absolutely.

Also, again, do not let this discussion block this PR. We can resolve this in a separate issue/PR.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/cockroachdb/cockroach/24851)*
<!-- Sent from Reviewable.io -->

rytaft commented:
Review status: 12 of 24 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/opt/optbuilder/testdata/project, line 290 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/24851#-LAFbovB9kmzczVqEbAz:-LAoHnCY9cNqMFMURt8V:b-xmbt1e) ([raw file](https://github.com/cockroachdb/cockroach/blob/963ac55427ff5fc9f1bc4e8550c6edf4062b2c17/pkg/sql/opt/optbuilder/testdata/project#L290)):*
<details><summary><i>Previously, andy-kimball (Andy Kimball) wrote…</i></summary><blockquote>

The name in the metadata is just a "tag" for our own internal debugging. It's not intended to be a fully qualified, fully accurate name. It's more important that these names are terse than that they're perfectly accurate). Unless we have a way to keep them terse in the common case where all tables are from the same db/catalog, I don't want to qualify them further.
</blockquote></details>

Ok I think the "fully-qualify-names" flag for name resolution tests should be a good compromise. I've made the change - please take another look.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/24851)*
<!-- Sent from Reviewable.io -->

rytaft commented:
@andy-kimball and @knz -- is this ok to merge? Testing with virtual schemas and additional name resolution tests are both listed on our spreadsheet of optimizer to-dos.

---

Review status: 11 of 24 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/24851#-:-LAxQaIR93goCU4NX8_m:buu7aoi)*
<!-- Sent from Reviewable.io -->

andy-kimball commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 11 of 24 files reviewed at latest revision, 1 unresolved discussion.

---

*[pkg/sql/opt/exec/execbuilder/testdata/stats, line 19 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/24851#-LAxSPRh6slywbgnPfA3:-LAxSPRh6slywbgnPfA4:bu22jqe) ([raw file](https://github.com/cockroachdb/cockroach/blob/e0998b786faf06906123577faf2510f385f6857b/pkg/sql/opt/exec/execbuilder/testdata/stats#L19)):*
> ```
>  ├── cost: 1000.00
>  ├── scan a
>  │    ├── columns: t.public.a.u:1(int) t.public.a.v:2(int) t.public.a.rowid:3(int!null)
> ```

Don't you want to suppress db/catalog names in these tests?

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/24851#-:-LAxT0ug9EzmhG1-j0iq:bnfp4nl)*
<!-- Sent from Reviewable.io -->

rytaft commented:
Thanks!

---

Review status: 11 of 31 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/opt/exec/execbuilder/testdata/stats, line 19 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/24851#-LAxSPRh6slywbgnPfA3:-LAxjlee4R9Ta-utGAGQ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/e0998b786faf06906123577faf2510f385f6857b/pkg/sql/opt/exec/execbuilder/testdata/stats#L19)):*
<details><summary><i>Previously, andy-kimball (Andy Kimball) wrote…</i></summary><blockquote>

Don't you want to suppress db/catalog names in these tests?
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/24851#-:-LAxk5nw4SpEUzCafsWB:b-ucfknh)*
<!-- Sent from Reviewable.io -->

rytaft commented:
bors r+

---

Review status: 11 of 31 files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/24851#-:-LAxrWcq7gV3TzILSoxi:b-ydi9ay)*
<!-- Sent from Reviewable.io -->
